tasks/internal/create-advisory-task/create-advisory-task.yaml

```
@@ -300,7 +300,7 @@ spec:
           cat "$ADVISORY_FILEPATH"
 
           # Ensure the created advisory file passes the advisory schema
-          check-jsonschema --schemafile schema/advisory.json "$ADVISORY_FILEPATH" 2> "$STDERR_FILE"
+          check-jsonschema --schemafile schema/advisory.json "$ADVISORY_FILEPATH" > "$STDERR_FILE"
```

Won’t redirecting using `>` cause the output to be hidden from the actual pipeline logs and only shown in results?
Shouldn’t we instead use something like `> >(tee "$STDERR_FILE")` so the logs are visible in the pipeline and captured for the results?

---

I agree with Sean and I would also make sure the stderr is captured as well (even if the usual errors are printed to stdout).
```suggestion
          check-jsonschema --schemafile schema/advisory.json "$ADVISORY_FILEPATH" 2>&1 | tee "$STDERR_FILE"
```

---

> I agree with Sean and I would also make sure the stderr is captured as well (even if the usual errors are printed to stdout).

Won't a pipe with a tee mask the actual exit status of check-jsonschema? It will return the exit code of tee instead, which's what I have found, so the script will pass.

I have been using this `> >(tee "$STDERR_FILE") 2>&1`  so it redirects stderr to the same place as stdout.
Maybe there's a better way to do it

---

I mean, if it fails, it writes to stdout. If it passes, it writes to stdout. I wouldn't want to spend too much time redirecting stderr for a command that never writes to stderr

---

> Won't a pipe with a tee mask the actual exit status of check-jsonschema? It will return the exit code of tee instead, that's what I have found, so the script will pass.

That would be a problem if `set -o pipefail` was not set. Which it is here.

I've never seen yours, but sure, if it works...

> I wouldn't want to spend too much time redirecting stderr for a command that never writes to stderr

Well, you're in luck because above you have two options to choose from :)

Here's an example where it does print to stderr:
```
check-jsonschema --schemafile ./schema/dataKeys.json ~/tmp/data-systems.json > /dev/null
Error: Unexpected Error building schema validator
FileNotFoundError: [Errno 2] No such file or directory: '/Users/mmalina/schema/dataKeys.json'
  in "/opt/homebrew/Cellar/check-jsonschema/0.33.1/libexec/lib/python3.13/site-packages/check_jsonschema/checker.py", line 56
  >>> return self._schema_loader.get_validator(
```

Of course, not likely to happen. But I don't see a reason why not do it. It's just about the extra `2>&1`. With or without it, I think we definitely want to still see the output in the internal task log.

---

ack, if y'all already have the solution for it, happy to use it. I haven't spent time reading the reviews all the way yet, but will be soon

---

Okay, confirmed Martin's thing works such that it still goes to STDERR_FILE and stdout and the exit code is nonzero if the command fails. I am going with that

---

done

---

cool, thanks for confirming. I'll let Sean resolve the thread, since he opened it.

---

I will say, the thing that stinks about `| tee` is now `command` in the trap is `tee`, not `check-jsonschema` so it says `tee` was the one that failed (which technically it is). But, since we now properly send the results of `check-jsonschema` it should be okay I guess